### PR TITLE
dmm_tools::dmm::Map::adjust_key_length() fix

### PR DESCRIPTION
**Pull request by DreamySkrell**
Currently, there is `Map::adjust_key_length()`, but it sets key length based on the amount of items in the dictionary. Which is fine if keys in order like 1-2-3-4-...-N, but it falls apart if the keys are not in order - which is valid for a dmm map.

If a map has keys like [1, 2, 3000], then, currently, `Map::adjust_key_length()` will set key length to 1, but the map will panic on save cause the 3000 key is bigger than max key for this key length (52).

This PR fixes `adjust_key_length` to work properly in that case, where it sets key length based on the biggest key in the map dictionary.

### CODE REVIEW

**Comment by SpaceManiac** [comment]
Perhaps this function should validate that the passed-in length is 1, 2, or 3.

Perhaps `adjust_key_length` should also be fixed!

**Comment by DreamySkrell** [comment]
> Perhaps this function should validate that the passed-in length is 1, 2, or 3.
> 
> Perhaps `adjust_key_length` should also be fixed!

Now thinking about it, fixing `adjust_key_length` would be better and safer, where it would iterate over all dict keys and get the largest key, and set key length based on that.

Even with the validation for `set_key_length` as written currently, the user could still give it 1 when there's too many keys, and it would panic on save.

**Comment by DreamySkrell** [comment]
Updated code and PR as above.

File: crates/dmm-tools/src/dmm.rs [hunk 1]
```diff
@@ -194,12 +194,15 @@ impl Map {
...
-            self.key_length = 3;
-        } else if self.dictionary.len() > 52 {
-            self.key_length = 2;
-        } else {
-            self.key_length = 1;
+        if let Some(max_key) = self.dictionary.keys().max() {
+            let max_key = max_key.0;
+            if max_key > 2704 {
+                self.key_length = 3;
+            } else if max_key > 52 {
```
Comment (SpaceManiac) [inline crates/dmm-tools/src/dmm.rs:201]: I think these two conditions should now be >= instead of just >

Comment (DreamySkrell) [reply crates/dmm-tools/src/dmm.rs]: addressed in most recent commit

**Pull request merged**